earth_enterprise/src/fusion/portableglobe/cutter/cgi-bin/globe_cutter_app.py

```diff
@@ -438,11 +473,13 @@ def AddJsonIcons(self, source, json):
       # Get JSON from the server.
       url = "%s/query?request=Icon&icon_path=icons/%s" % (source, icon)
       try:
-        fp = urllib.urlopen(url)
-        fpw = open("%s/%s" % (self.icons_dir, icon), "w")
-        fpw.write(fp.read())
-        fp.close()
-        fpw.close()
+        data, code = self.HttpGet(url)
+        if code == 200:
+          fpw = open("%s/%s" % (self.icons_dir, icon), "w")
+          fpw.write(data)
+          fpw.close()
```

Comment:
Also here can we use the `with` clause to ensure close is called?

https://www.pythonforbeginners.com/files/with-statement-in-python